nodes/src/nodes/multi_agent/orchestrator.py

```
@@ -0,0 +1,549 @@
+            for future in as_completed(futures):
+                try:
+                    results.append(future.result())
+                except Exception as exc:
+                    task = futures[future]
+                    results.append(
+                        AgentResult(
+                            agent_name=task.assigned_agent,
+                            task_id=task.id,
+                            error=str(exc),
+                        )
+                    )
```

Ruff (0.15.7)

[warning] `try`-`except` within a loop incurs performance overhead

(PERF203)